moto/sagemaker/models.py

```diff
@@ -0,0 +1,364 @@
+from __future__ import unicode_literals
+from datetime import datetime
+
+from .exceptions import MissingModel
+from moto.ec2 import ec2_backends
+from moto.ecr.models import BaseObject
+from moto.core import BaseBackend
+from moto.core.exceptions import RESTError
+from moto.sagemaker import validators
+from moto.sts.models import ACCOUNT_ID
+
+
+class Model(BaseObject):
+    def __init__(
+        self,
+        region_name,
+        model_name,
+        execution_role_arn,
+        primary_container,
+        vpc_config,
+        containers=[],
+        tags=[],
+    ):
+        self.ModelName = model_name
+        self.creation_time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        self.Containers = containers
+        self.Tags = tags
+        self.EnableNetworkIsolation = False
+        self.VpcConfig = vpc_config
+        self.PrimaryContainer = primary_container
+        self.ExecutionRoleArn = execution_role_arn or "arn:test"
+        self.ModelArn = self.arn_for_model_name(self.ModelName, region_name)
+
+    @property
+    def response_object(self):
+        response_object = self.gen_response_object()
+        return {
+            k: v for k, v in response_object.items() if v is not None and v != [None]
+        }
+
+    @property
+    def response_create(self):
+        return {"ModelArn": self.ModelArn}
+
+    @staticmethod
+    def arn_for_model_name(model_name, region_name):
```

Comment:
`model_name` and `region_name` should both be part of the Model-instance, so this can be a non-static method without any args. (Same as `FakeSagemakerNotebookInstance().arn`)

Reply:
Bert, the reason for the _staticmethod arn_for_model_name()_ is to be able to format an arn for a resource _that doesn't exist_, typically to be included in an error message, so we can pretty closely replicate the actual error messages returned by boto3 against live AWS.  You'll see this done in SageMakerModelBackend.describe_model().

Rather than duplicate that logic for formatting actual resource object's arns, I use this staticmethod also to initialize a resource object's actual arn attribute (or implement its property).